src/specBuilder/marks/markUtils.ts

```diff
@@ -273,3 +300,127 @@ export const getVoronoiPath = (children: MarkChildElement[], dataSource: string,
 		},
 	],
 });
+/**
+ * the signal that triggers the opacity ease in and out
+ * @param opacityValue
+ * @returns { signal: string}
+ */
+//TODO: Add tests
+export const getHighlightOpacityAnimationValue = (opacityValue: { signal: string } | { value: number }): { signal: string }  => {
+	if ('signal' in opacityValue) {
+		return { signal: `max(1-rscColorAnimation, ${opacityValue.signal} / ${HIGHLIGHT_CONTRAST_RATIO})` }
+	}
+	return { signal: `max(1-rscColorAnimation, ${opacityValue.value} / ${HIGHLIGHT_CONTRAST_RATIO})`}
```

Comment:
This works fine, but since only the variable in our string is different, we could avoid some duplicate code by assigning the right one to a variable. This also means it's no longer necessary to look at both lines to determine what's different between them.

```suggestion
        const opacity = 'signal' in opacityValue ? opacityValue.signal : opacityValue.value

	return { signal: `max(1-rscColorAnimation, ${opacity} / ${HIGHLIGHT_CONTRAST_RATIO})` }

```